src/legacy/core_plugins/console/np_ready/public/lib/token_iterator/token_iterator.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { TokenIterator, Position, Token, TokensProvider } from '../../interfaces';
+
+function colIsInTokenRange(column: number, token: Token) {
+  return (
+    token.position.column <= column && token.position.column + (token.value.length - 1) >= column
+  );
```

Comment:
Just a small suggestion, but returning early could make this a little easier for readers to reason about:

```ts
function colIsInTokenRange(column: number, token: Token) {
  if (column < token.position.column) {
    return false;
  }

  if (column > token.position.column + token.value.length - 1) {
    return false;
  }

  return true;
}
```